overlord/configstate/configcore/kernel.go

```diff
@@ -64,6 +64,12 @@ func validateCmdlineParamsAreAllowed(st *state.State, devCtx snapstate.DeviceCon
 	if err != nil {
 		return err
 	}
+	if gd == nil {
+		// no gadget data currently, so we cannot check against the allow-list
+		// this can happen on classic systems
+		return fmt.Errorf("changing the kernel command line is not supported on a classic system")
```

Comment:
hmm wouldn't there be a way to build classic system with a gadget and a model though?

Reply:
but hardly matter doesn't it? - it would still hit the error once the task executes in handlers_gadget